### PR TITLE
Ignore classes by regular expression

### PR DESCRIPTION
**Pull request by mbellomo**
Update of my pull request for ning's #23.  This allows you to ignore classes by use of a regular expression.

Can be used as a solution for one of these depending on interpretation.
Issue #5 
Issue #6

### CODE REVIEW

**Comment by mbellomo** [comment]
Is there a chance this is going to get accepted?  Trying to upgrade some of my code and will tag off my fork if needed, but would rather not have to do so.

**Comment by hgschmie** [comment]
There is an excellent chance that this gets merged. I am a bit pressed for time but if you need it, I will find time this week.

Thanks for the patch!

**Comment by hgschmie** [comment]
You can increase the chances of me merging this tremendously by writing a couple of integration tests. see https://github.com/basepom/duplicate-finder-maven-plugin/wiki/Writing%20Integration%20Tests for a guide. Otherwise, I would have to write some and, being more time consuming for me, will push merging out.

File: src/main/java/org/basepom/mojo/duplicatefinder/classpath/ClasspathDescriptor.java [hunk 1]
```diff
@@ -73,6 +73,12 @@
         ".*overview\\.html$"
         ));
 
+    private static final MatchPatternPredicate DEFAULT_IGNORED_CLASS_PREDICATE = new MatchPatternPredicate(Arrays.asList(
+        // this regex matches nothing at all, just need something as a place holder
```
Comment (hgschmie) [inline src/main/java/org/basepom/mojo/duplicatefinder/classpath/ClasspathDescriptor.java]: huh, what? This should either be empty or, if MatchPatternPredicate for whatever reason does not support an empty array, it should be fixed.

**Comment by hgschmie** [comment]
So the patch looks pretty reasonable. Any chance to whip up some integration tests? I will not merge function changes without integration tests. Also, it would be great to add docs (see https://github.com/basepom/duplicate-finder-maven-plugin/wiki/Contributing%20Documentation) on how to do that).

I would love to have this functionality in the mainline plugin but it that means I have to spend hours of doing the "boring pieces" (write tests and documentation), it will either have to wait until I find time for that or you need to contribute it with the code.

**Comment by mbellomo** [comment]
My time's been bad too, let me put some tests together.  I can update the docs too, no worries.  I'll update the pull request when it's done.  Sorry for the delay, been busy ;)

File: src/main/java/org/basepom/mojo/duplicatefinder/classpath/ClasspathDescriptor.java (hunk 1 above)
Comment (mbellomo) [inline src/main/java/org/basepom/mojo/duplicatefinder/classpath/ClasspathDescriptor.java]: It doesn't support empty arrays, but I've changed my code (included in next push) to use your inner class predicate as the default, which avoids the need for this.

**Comment by mbellomo** [comment]
@hgschmie - this should be ready for you now

**Comment by hgschmie** [comment]
Excellent. Just in time for a new release.

**Comment by hgschmie** [comment]
rebased and merged.

**Comment by mbellomo** [comment]
Thanks!

**Pull request closed without merging**